Source/SeaInk.Core/Entities/Tables/GoogleTable.cs

```diff
@@ -0,0 +1,338 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Services;
+using Google.Apis.Sheets.v4;
+using Google.Apis.Sheets.v4.Data;
+using Google.Apis.Util.Store;
+using SeaInk.Core.Models.Tables;
+using SeaInk.Core.Models.Tables.Exceptions;
+
+namespace SeaInk.Core.Entities.Tables
+{
+    /// <inheritdoc cref="ITable"/>
+    public class GoogleTable : ITable
+    {
+        private static readonly string[] Scopes = {SheetsService.Scope.Spreadsheets};
+        private const string ApplicationName = "Sea Ink";
+
+        private string SpreadsheetId { get; set; }
+        private SheetsService Service { get; set; }
+
+        public int SheetCount => GetSpreadsheet().Sheets.Count;
+        
+        
+        public GoogleTable()
+        {
+            UserCredential credential;
+            using (var stream =
+                new FileStream("credentials.json", FileMode.Open, FileAccess.Read))
+            {
+                const string credPath = "token.json";
+                credential = GoogleWebAuthorizationBroker.AuthorizeAsync(
+                    GoogleClientSecrets.Load(stream).Secrets,
+                    Scopes,
+                    "user",
+                    CancellationToken.None,
+                    new FileDataStore(credPath, true)).Result;
+                Console.WriteLine("Credential file saved to: " + credPath);
+            }
```

Comment:
Теперь можно делать так, тогда не будет замусоривания типа `UserCredential credential;`
```suggestion
            using var stream = new FileStream("credentials.json", FileMode.Open, FileAccess.Read);
            const string credPath = "token.json";
            var credential = GoogleWebAuthorizationBroker.AuthorizeAsync(
                GoogleClientSecrets.Load(stream).Secrets,
                Scopes,
                "user",
                CancellationToken.None,
                new FileDataStore(credPath, true)).Result;
            Console.WriteLine("Credential file saved to: " + credPath);
            
            Service = new SheetsService(new BaseClientService.Initializer
            {
                HttpClientInitializer = credential,
                ApplicationName = ApplicationName,
            });
```